core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -162,6 +163,51 @@ public String defaultHostname() {
         }
     }
 
+    /**
+     * Returns the primary {@link ServerPort}'s port that this {@link Server} is listening to.
+     *
+     * @throws IllegalStateException if no available active ports
+     */
+    public int activeLocalPort() {
+        synchronized (activePorts) {
+            ServerPort serverPort = Iterables.getFirst(activePorts.values(), null);
```

Comment:
You should specify a predicate here as well. How about creating a `private static` method like this:

```java
private static boolean isLocalPort(ServerPort port, @Nullable SessionProtocol protocol) {
    ...
}
```

..?

Reply:
I think you did not address this issue, @zacscoding :wink: